hazelcast/src/test/java/com/hazelcast/internal/serialization/impl/AbstractSerializationServiceTest.java

```diff
@@ -0,0 +1,172 @@
+        abstractSerializationService.toBytes(new StringBuffer());
+    }
+
+    @Test
+    public void testToObject_ServiceInactive() throws Exception {
+        expectedException.expect(HazelcastSerializationException.class);
+        expectedException.expectCause(Is.is(IsInstanceOf.<Throwable>instanceOf(HazelcastInstanceNotActiveException.class)));
+
+        abstractSerializationService.register(StringBuffer.class, new StringBufferSerializer(false));
+        Data data = abstractSerializationService.toData(new StringBuffer());
+        abstractSerializationService.destroy();
+        abstractSerializationService.toObject(data);
+    }
+
+    @Test(expected = HazelcastSerializationException.class)
+    public void testWriteObject_serializerFail() throws Exception {
+        abstractSerializationService.register(StringBuffer.class, new StringBufferSerializer(true));
+        BufferObjectDataOutput out = abstractSerializationService.createObjectDataOutput();
+        abstractSerializationService.writeObject(out, new StringBuffer());
+    }
+
+    @Test
+    public void testReadObject_ServiceInactive() throws Exception {
+        expectedException.expect(HazelcastSerializationException.class);
+        expectedException.expectCause(Is.is(IsInstanceOf.<Throwable>instanceOf(HazelcastInstanceNotActiveException.class)));
+
+        abstractSerializationService.register(StringBuffer.class, new StringBufferSerializer(false));
+        Data data = abstractSerializationService.toData(new StringBuffer());
+        abstractSerializationService.destroy();
+
+        BufferObjectDataInput in = abstractSerializationService.createObjectDataInput(data);
+        in.position(HeapData.TYPE_OFFSET);
+        abstractSerializationService.readObject(in);
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void testRegister_nullType() throws Exception {
+        abstractSerializationService.register(null, new StringBufferSerializer(true));
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void testRegister_typeIdNegative() throws Exception {
+        StringBufferSerializer serializer = new StringBufferSerializer(true);
+        serializer.typeId = -10000;
+        abstractSerializationService.register(StringBuffer.class, serializer);
+    }
+
+    @Test(expected = IllegalStateException.class)
+    public void testGlobalRegister_doubleRegistration() throws Exception {
+        abstractSerializationService.registerGlobal(new StringBufferSerializer(true));
+        abstractSerializationService.registerGlobal(new StringBufferSerializer(true));
+    }
+
+    @Test(expected = IllegalStateException.class)
+    public void testGlobalRegister_alreadyRegisteredType() throws Exception {
+        abstractSerializationService.register(StringBuffer.class, new StringBufferSerializer(true));
+        abstractSerializationService.registerGlobal(new TheOtherGlobalSerializer(true));
+    }
```

Comment:
I'm not sure why this should throw an exception, the global serializer should be separate from custom serializers, even if it extends the class?